packages/gatsby/src/utils/babel-loader.js

```diff
@@ -48,11 +56,27 @@ module.exports = babelLoader.custom(babel => {
           ...options,
         },
       }
+
+      customOptionsCache.set(stage, toReturn)
+
+      return toReturn
```

Comment:
Might be overkill and I didn't check this in isolation - but we only have 2 stages per session, so why not skip repeatable work